core/plugins/org.fusesource.ide.camel.model.service.v2_18_1/src/org/fusesource/ide/camel/model/service/internal/CamelService.java

```diff
@@ -0,0 +1,201 @@
+/******************************************************************************* 
+ * Copyright (c) 2015 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.fusesource.ide.camel.model.service.internal;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Map;
+
+import org.apache.camel.catalog.CamelCatalog;
+import org.apache.camel.catalog.DefaultCamelCatalog;
+import org.apache.camel.converter.TimePatternConverter;
+import org.apache.camel.util.URISupport;
+import org.fusesource.ide.camel.model.service.core.CamelSchemaProvider;
+import org.fusesource.ide.camel.model.service.core.ICamelManagerService;
+import org.fusesource.ide.camel.model.service.core.adopters.CamelModelLoader;
+import org.fusesource.ide.camel.model.service.core.adopters.XmlCamelModelLoader;
+import org.fusesource.ide.camel.model.service.core.catalog.CamelModel;
+import org.fusesource.ide.camel.model.service.core.catalog.CamelModelFactory;
+
+/**
+ * @author lhein
+ */
+public class CamelService implements ICamelManagerService {
+	
+	private static final boolean ENCODE_DEFAULT = false;
+	
+	private CamelModelLoader loader;
+	private CamelCatalog catalog;
+
+	/* (non-Javadoc)
+	 * @see org.fusesource.ide.camel.model.service.core.ICamelManagerService#getCamelModel()
+	 */
+	@Override
+	@Deprecated
+	public CamelModel getCamelModel() {
+		return getCamelModel(CamelModelFactory.RUNTIME_PROVIDER_KARAF);
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.fusesource.ide.camel.model.service.core.ICamelManagerService#getCamelModel(java.lang.String)
+	 */
+	@Override
+	public CamelModel getCamelModel(String runtimeProvider) {
+		this.loader = new XmlCamelModelLoader();
```

Comment:
the loader class attribute is now used only in this method, so attribute can be removed. Unless we want to keep a cache for it